### ISSUE TITLE
Add latex `import` package support for complex project structure

### ISSUE THREAD

**Issue by tomasyany** [lervag/vimtex#241]
See [this question](http://vi.stackexchange.com/questions/4983/latex-plugin-for-vim-that-supports-files-organized-in-multiple-directories) on SO which explains a situation where `vimtex` seems not to work.

**Comment by lervag** [comment]
Ok. I've looked at the [documentation](http://ctan.uib.no/macros/latex/contrib/import/import.pdf) of the import package. If I understand correctly, the point is to allow for instance the chapter to use relative imports.

Vimtex already supports the usage of `\include` and `\input`. These require paths relative to the main file. To add support for the import package, I must change the parsers to allow the `\subimportant` (and possibly other) command(s). I'm just curious, to me it seems your particular structure as described in the SO question is unnecessarily complicated. You could accomplish a similar setup using only includes and inputs...

**Comment by tomasyany** [comment]
I was wondering why did I use the `import` package, and the reason is that it is the only way to have visually clean imports. If I didn't have a main file which imports sub-main files through `\subimport{sub-main-file}`, which import the final files, then in the last `\input` I would have to write `\input{sub-folder/file.tex}` instead of simply `\input{file.tex}`, `file.tex` being present under the `sub-folder` folder. 

In other words, the first `\subimport` allows to avoid the redundance of writing the sub-folder name for each sub-sub-file import.

**Comment by lervag** [comment]
Just noticed that this seems related to #208.

**Comment by lervag** [comment]
I will look at this when I get the time. I think I should be able to add support for this and at the same time adopt the suggestions for support for Chapterfolder (as in #208). I can't promise that I will manage to get it done very quickly; I will do it when I get the time. This requires quite a few changes and updates, as there are quite a lot of places where we want to recursively parse the entire project (and thus follow the imports).

**Comment by lervag** [comment]
By the way: It would be very helpful if you could provide a very small example. That is, if you could provide a small tex example that uses the `import` package and that has a structure similar to the one you want. Please keep the content as minimal as possible. I will include the files as part of the vimtex repository for future reference and for testing purposes.

I would think that this could be done simply by stripping down your current project as much as possible, removing most of the files except a few, removing the content and all packages that are not relevant.

**Comment by lervag** [comment]
As you can see, I've added a test file for this issue. However, I am not sure if it is a good test file, so I would be happy if you could take a quick look (see `$VIMTEX/test/issue/241`).

Am I correct that the only relevant command here is the following?

```tex
\subimport{subdir/}{file}
```

And if I understand correctly, the point of this is to allow relative imports in `file`. Correct?

**Comment by tomasyany** [comment]
Sorry, I didn't enter github in a while. Yes, your test files are correct. The point is for `vimtex` to recognize relative imports through the `import` package

**Comment by lervag** [comment]
Sorry for taking so long! I've pushed a first attempt at solving this issue. I think it should work, at least it works on the minimal example under the `vimtex/test/issue/241` folder.
